service/lib/agama/manager.rb

```diff
@@ -71,7 +72,9 @@ def initialize(config, logger)
       @installation_phase = InstallationPhase.new
       @service_status_recorder = ServiceStatusRecorder.new
       @service_status = DBus::ServiceStatus.new.busy
+
       on_progress_change { logger.info progress.to_s }
+      Ipmi.setup(logger)
```

Comment:
ideally here `@ipmi = Ipmi.new(logger)` and below like `@ipmi.started`.

Reply:
ok i'll destroy the singleton ;-)